numba/targets/arrayobj.py

```diff
@@ -3208,9 +3220,10 @@ def make_array_nditer(context, builder, sig, args):
     res = nditer._getvalue()
     return impl_ret_borrowed(context, builder, nditerty, res)
 
+
 @lower_builtin('iternext', types.NumpyNdIterType)
 @iternext_impl(RefType.BORROWED)
-def iternext_numpy_ndindex(context, builder, sig, args, result):
+def iternext_numpy_ndindex_borrowed(context, builder, sig, args, result):
```

Comment:
all the refs are borrowed, the difference here is that this is an iterator for the `nditer` type, whereas the above is for the `ndindex` type.